### PR TITLE
Add-Dropdown-on-Dashboard

### PR DESCRIPTION
**Pull request by Harshvardhan80**
Targets #3080 

### Description
Added dropdown for officelocation on dashboard page in which the centre name appears.
After selecting the name of the centre, it is store in the usermeta table.

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: resources/views/home.blade.php [hunk 1]
```diff
@@ -29,29 +43,29 @@
...
     </div>
 </div>
+
 @includeWhen($book, 'knowledgecafe.library.books.show_nudge_modal')
 
 @endsection
 
 @section('inline_js')
-new Vue({ el: '#home_page'});
+    new Vue({ el: '#home_page' });
```
Comment (Ayush8923) [inline resources/views/home.blade.php:70]: Can you please remove the extra spaces? And please update all places.

File: resources/views/home.blade.php [hunk 2]
```diff
@@ -17,9 +17,23 @@
...
+        <span>Your FTE: <b class="{{ auth()->user()->ftes['main'] < 1 ? 'text-danger' : 'text-success' }}">{{ auth()->user()->ftes['main'] }}</b></span>
     </div>
+
+    <form method="POST" action="{{ route('storeDropdownValue') }}">
+        @csrf
+        <div class="dropdown">
+            <select class="btn bg-light" name="selectedValue" onchange="this.form.submit()">
+                @foreach($centre as $location)
+                    <option value="{{ $location->centre_name }}">{{ $location->centre_name }}</option>
+                @endforeach
```
Comment (Ayush8923) [inline resources/views/home.blade.php:31]: Can we write like this?
```suggestion
                @foreach($centres as $centre)
                    <option value="{{ $centre->centre_name }}">{{ $centre->centre_name }}</option>
                @endforeach
```

File: app/Http/Controllers/HomeController.php [hunk 3]
```diff
@@ -24,14 +27,16 @@ public function __construct()
     public function index()
     {
         $unreadBook = (session('disable_book_suggestion')) ? null : Book::getRandomUnreadBook();
+        $centre = OfficeLocation::orderBy('centre_name', 'asc')->get();
 
-        return view('home')->with(['book' => $unreadBook]);
+        return view('home')->with(['book' => $unreadBook, 'centre' => $centre]);
```
Comment (Ayush8923) [inline app/Http/Controllers/HomeController.php:32]: Please update the all places.
```suggestion
        return view('home')->with(['book' => $unreadBook, 'centres' => $centres]);
```

File: routes/web.php [hunk 4]
```diff
@@ -22,6 +22,7 @@
 Auth::routes();
 
 Route::get('home', 'HomeController@index')->name('home');
+Route::post('/store-dropdown-value', 'HomeController@storeDropdownValue')->name('storeDropdownValue');
```
Comment (Ayush8923) [inline routes/web.php:25]: Can you please use an understandable name instead of `store-dropdown-value`?

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Left some comments. 
Please check.

File: routes/web.php [hunk 5]
```diff
@@ -22,6 +22,7 @@
 Auth::routes();
 
 Route::get('home', 'HomeController@index')->name('home');
+Route::post('/location', 'HomeController@storeDropdownValue')->name('storeDropdownValue');
```
Comment (Ayush8923) [inline routes/web.php:25]:
```suggestion
Route::post('/current-employee-location', 'HomeController@storeDropdownValue')->name('employee.location');
```

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Testing Feedback
- [ ] When I choose the current location the current location is saved in the database but it's not reflected in the UI Dropdown. So current location also reflecting in the Dropdown.
**Please take a reference below screenshot.**
image

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Testing Feedback
- [x] When I choose the current location the current location is saved in the database but it's not reflected in the UI Dropdown. So current location also reflecting in the Dropdown.
**Please take a reference below screenshot.**
image

File: routes/web.php [hunk 6]
```diff
@@ -22,6 +22,7 @@
 Auth::routes();
 
 Route::get('home', 'HomeController@index')->name('home');
+Route::post('/current-employee-location', 'HomeController@storeDropdownValue')->name('employee.location');
```
Comment (Ayush8923) [inline routes/web.php:25]:
```suggestion
Route::post('/current-employee-location', 'HomeController@storeEmployeeLocation')->name('employee.location');
```

File: app/Http/Controllers/HomeController.php [hunk 7]
```diff
@@ -58,4 +71,18 @@ public function getUserGroups($email)
 
         return $userGroups;
     }
+
+    public function storeDropdownValue(Request $request)
```
Comment (Ayush8923) [inline app/Http/Controllers/HomeController.php:75]:
```suggestion
    public function storeEmployeeLocation(Request $request)
```

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Left one small comment.

File: app/Http/Controllers/HomeController.php [hunk 8]
```diff
@@ -58,4 +71,18 @@ public function getUserGroups($email)
 
         return $userGroups;
     }
+
+    public function storeEmployeeLocation(Request $request)
+    {
+        $validatedData = $request->validate([
+            'Centre' => 'required',
```
Comment (GauravGusain98) [inline app/Http/Controllers/HomeController.php:78]: The name should be in small case. Also read about [exist rule in Laravel](https://laravel.com/docs/10.x/validation#basic-usage-of-exists-rule)
```suggestion
            'centre_name' => 'required | exists:office_locations,centre_name',
```

File: app/Http/Controllers/HomeController.php [hunk 9]
```diff
@@ -58,4 +71,18 @@ public function getUserGroups($email)
...
+        ]);
+
+        $userMeta = UserMeta::updateOrCreate(
+            ['user_id' => auth()->user()->id, 'meta_key' => 'office_location'],
+            ['meta_value' => $request->Centre]
```
Comment (GauravGusain98) [inline app/Http/Controllers/HomeController.php:83]:
```suggestion
            ['meta_value' => $request->centre_name]
```

File: resources/views/home.blade.php [hunk 10]
```diff
@@ -20,6 +20,23 @@
     <div class="mb-2">
         <span>Your FTE: <b class="{{ auth()->user()->ftes['main'] < 1 ? 'text-danger' : 'text-success' }}">{{auth()->user()->ftes['main'] }}</b> </span>
     </div>
+
+    <form method="POST" action="{{ route('employee.location') }}">
+        @csrf
+        <div class="dropdown">
+            <span>Office Location</span>
+            <select class="btn bg-light text-left" name="Centre" onchange="this.form.submit()">
```
Comment (GauravGusain98) [inline resources/views/home.blade.php:28]:
```suggestion
            <select class="btn bg-light text-left" name="centre_name" onchange="this.form.submit()">
```

File: resources/views/home.blade.php [hunk 11]
```diff
@@ -20,6 +20,23 @@
     <div class="mb-2">
         <span>Your FTE: <b class="{{ auth()->user()->ftes['main'] < 1 ? 'text-danger' : 'text-success' }}">{{auth()->user()->ftes['main'] }}</b> </span>
     </div>
+
+    <form method="POST" action="{{ route('employee.location') }}">
+        @csrf
+        <div class="dropdown">
+            <span>Office Location</span>
```
Comment (GauravGusain98) [inline resources/views/home.blade.php:27]:
```suggestion
            <span>Your office location</span>
```

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
LGTM

File: app/Http/Controllers/HomeController.php [hunk 12]
```diff
@@ -24,14 +27,24 @@ public function __construct()
     public function index()
     {
         $unreadBook = (session('disable_book_suggestion')) ? null : Book::getRandomUnreadBook();
+        $centres = OfficeLocation::orderBy('centre_name', 'asc')->get();
 
-        return view('home')->with(['book' => $unreadBook]);
+        $selectedLocation = UserMeta::where('user_id', auth()->user()->id)
+            ->where('meta_key', 'office_location')
+            ->value('meta_value');
```
Comment (GauravGusain98) [inline app/Http/Controllers/HomeController.php:34]: Can we make a fucntion in user Model such that we can easily fetch office_location like 

auth()->user()->office_location

File: app/Http/Controllers/HomeController.php [hunk 13]
```diff
@@ -58,4 +71,18 @@ public function getUserGroups($email)
 
         return $userGroups;
     }
+
+    public function storeEmployeeLocation(Request $request)
+    {
+        $validatedData = $request->validate([
+            'centre_name' => 'required | exists:office_locations,centre_name',
```
Comment (GauravGusain98) [inline app/Http/Controllers/HomeController.php:78]: Can you share what this `exists:office_locations,centre_name` do?

File: routes/web.php [hunk 14]
```diff
@@ -22,6 +22,7 @@
 Auth::routes();
 
 Route::get('home', 'HomeController@index')->name('home');
+Route::post('/current-employee-location', 'HomeController@storeEmployeeLocation')->name('employee.location');
```
Comment (GauravGusain98) [inline routes/web.php:25]:
```suggestion
Route::post('/current-employee-location', 'HomeController@storeEmployeeLocation')->name('update.employee.location');
```

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]

**Review by GauravGusain98 — APPROVED** [review]
LGTM

**Review by Ayush8923 — APPROVED** [review]

**Pull request merged**